File: frontend/src/app/dtos/userReported.dto.ts
import { ReviewReportedDTO } from "./reviewReported.dto";




export interface UserReportedDTO {
    id?: number,
    name: string,
    surname: string,
    email: string,
    address: string,
    banned: boolean,
    reports: number,
    imageUrl: String,
    roles: String[],
    showDetails?: boolean,
    showReports?: boolean,
    reviews: ReviewReportedDTO[],
}

File: backend/src/main/java/es/codeurjc/webapp14/dto/UserReportedDTO.java
package es.codeurjc.webapp14.dto;

import java.util.List;


public record UserReportedDTO (
    Long id,
    String name,
    String surname,
    String email,
    String address,
    boolean banned,
    int reports,
    String imageUrl,
    List<String> roles,
    List<ReviewReportedDTO> reviews
) {}
